### DISCUSSION TITLE
Project provisioning

### DISCUSSION THREAD

**Discussion by henningst** [Talentech/EvaluationApi#18] [Ideas]
To improve the user experience, it is desired to let the partner be notified about new recruitment projects in the ATS before the first invitation is triggered. It's also desired to let the ATS user easily edit the project specific template by allow the user to navigate from the ATS to the partner UI.

A draft outlining the feature can be found in the `drafts/project-provisioning` branch here:
https://github.com/Talentech/EvaluationApi/blob/drafts/project-provisioning/docs/project-provisioning.md

It was decided on 2022-09-06 to put further work on this feature on hold until.

**Comment by perbergland** [comment]
This will need significant rework for us (Refapp) to use

1. Responses cannot be cached only on tenantId+projectId+evaluationFormId but need to have triggeredBy as well since different users will see different results. They also need a low TTL (e.g. 10 minutes) since changes to forms and other settings can happen. We suggest not caching at all.
2. Creating a project needs access to all customData fields in addition to TriggeredBy so a new API call is better than trying to stuff all information into a URL. We do not think that an editUrl property is a good solution.
A /createRemoteProject endpoint that is idempotent would be an option, and then that endpoint could return one or more urls to expose in the ATS UI. Unsolved here is what languages to use, maybe we should send out UI texts in all languages configured for the account in Refapp.
3. This is NOT only about editing forms - it is about creating a project in the external system and THEN maybe edit project settings for that project. Those settings include form selection or editing the form itself. You should steer away from having this form-centric view of integrations.
4. We would like to also see the user’s UI language sent into the /evaluationForms API so that we can render UI strings in the correct locale (this should then be added to the cache key if you want to use a cache).
